controllers/numaresourcesoperator_controller.go

```diff
@@ -152,6 +152,10 @@ func (r *NUMAResourcesOperatorReconciler) Reconcile(ctx context.Context, req ctr
 		return r.updateStatus(ctx, instance, status.ConditionDegraded, validation.NodeGroupsError, err)
 	}
 
+	if err := validation.MultipleMCPsPerTree(instance.Annotations, trees); err != nil {
+		return r.updateStatus(ctx, instance, status.ConditionDegraded, validation.NodeGroupsError, err)
+	}
```

Comment:
almost there. The problem is this should be a soft degradation, because this configuration was allowed till last version. Degrading and do nothing, because of pre-validation, is just liiiiitle better than rejecting the setting straight away.

What we should do is: check if there are multiple trees per pool (done). Then proceed to reconcile everything else as usual.

Only at the very end, if the reconciliation succeeded BUT we had multiple trees per pool, force the state to degraded.

This way the system keep working like last release, but the user get a little annoyance to expose the wrong configuration.

Reply:
sounds better. Done, Thanks!